src/main/kotlin/no/nav/familie/ef/sak/task/KonsistensavstemmingTask.kt

```diff
@@ -0,0 +1,67 @@
+package no.nav.familie.ef.sak.task
+
+import no.nav.familie.ef.sak.integration.OppdragClient
+import no.nav.familie.ef.sak.repository.KonsistensavstemmingRepository
+import no.nav.familie.ef.sak.repository.domain.Stønadstype
+import no.nav.familie.ef.sak.økonomi.tilKlassifisering
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.oppdrag.KonsistensavstemmingRequest
+import no.nav.familie.kontrakter.felles.oppdrag.OppdragIdForFagsystem
+import no.nav.familie.prosessering.AsyncTaskStep
+import no.nav.familie.prosessering.TaskStepBeskrivelse
+import no.nav.familie.prosessering.domene.Task
+import no.nav.familie.prosessering.domene.TaskRepository
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.time.LocalDateTime
+import com.fasterxml.jackson.module.kotlin.readValue
+
+@Service
+@TaskStepBeskrivelse(taskStepType = KonsistensavstemmingTask.TYPE, beskrivelse = "Utfører konsistensavstemming mot økonomi.")
+class KonsistensavstemmingTask(
+        private val oppdragClient: OppdragClient,
+        private val taskRepository: TaskRepository,
+        private val konsistensavstemmingRepository: KonsistensavstemmingRepository,
+) : AsyncTaskStep {
+
+    val logger: Logger = LoggerFactory.getLogger(this.javaClass)
+
+    override fun doTask(task: Task) {
+        val payload = objectMapper.readValue<KonsistensavstemmingPayload>(task.payload)
+        konsistensavstemmingRepository.finnKonsistensavstemmingMedDatoIdag(stønadstype = payload.stønadstype)
+                ?.let { utførKonsistensavstemming(payload.stønadstype) }
+
+    }
+
+    override fun onCompletion(task: Task) {
+        val payload = objectMapper.readValue<KonsistensavstemmingPayload>(task.payload)
+        opprettNyTask(triggerTid = LocalDate.now().plusDays(1).atTime(8, 0), stønadstype = payload.stønadstype)
```

Comment:
Slik jeg tolker dette kommer den kjøre en ny task hver dag, men egentlige gjør vi kun denne avstemmingen en gang per måned?

Reply:
Det er riktigt.

Reply:
En möjlig förbättring hade vart att hämta nästa avstemmingensdato och sätta triggerTiden utifrån det

Reply:
Eller ær det møjligt att opprette Tasks direkt baserat på datumen som skickas in? 
Samtidigt hade det kanskje varit mer ryddig om man opprettet disse først i en tabell, som enkelt går å endre, så opprettes tasks når det trengs basert på den tabellen?

Reply:
Hm intressant! Tanken min var att GrensesnittavstemmingTask og KonsistenavstemmingTask er to like tasks. Enda skillnaden att grensesnittavstemming skal ske på virkedager medan konsistensavstemming ska skje på dager som økonomi-teamet har bestæmt. Så jag såg ingen grund till att implementera disse på forskjellige måter. Men er absolut øppen før andra løsningar :)

Reply:
Men jag gillade den idéen med  opprette Tasks direkt baserat på datumen som skickas in så kan testa ut den :D

Reply:
De kan nog være 2 likende tasks, men de kaller ulike metoder(selve avstemmingen)?
👍